Sources/ChessKit/Position.swift

```diff
@@ -29,6 +29,14 @@ public struct Position: Equatable, Sendable {
     /// en passant.
     var enPassant: EnPassant?
 
+    /// Conforming Position class to Hashable
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(pieceSet)
+        hasher.combine(sideToMove)
+        hasher.combine(legalCastlings)
+        hasher.combine(enPassant)
+    }
+    
```

Comment:
If `Clock` is made `Hashable` this won't be necessary either and will happen automatically